connect/src/spirc.rs

```diff
@@ -817,6 +827,17 @@ impl SpircTask {
             // Check for malformed gid
             let tracks_len = self.state.get_track().len() as u32;
             let mut track_ref = &self.state.get_track()[index as usize];
+            if track_ref.get_gid().len() != 0
```

Comment:
Would this be better if it's a bit more explicit? Something like:
```
            if track_ref.get_uri().starts_with("spotify:episode:")
            {
                SpotifyId::from_rawURI(track_ref.get_uri()).unwrap()
            }
            else
            {
                while track_ref.get_gid().len() != 16 {
                    warn!(
                        "Skipping track {:?} at position [{}] of {}",
                        track_ref.get_uri(),
                        index,
                        tracks_len
                    );
                    index = if index + 1 < tracks_len { index + 1 } else { 0 };
                    track_ref = &self.state.get_track()[index as usize];
                }
                SpotifyId::from_raw(track_ref.get_gid()).unwrap()
            }
```
Then we can keep the old logic for handling malformed GIDs, added in #305 I'm not sure exactly what was being avoided there but a blank GID may well be one of the cases.